include/seqan3/range/decorator/gap_decorator_anchor_set.hpp

```diff
@@ -0,0 +1,537 @@
+         * \returns A boolean flag indicating whether the gap starting position
+         *          of \p lhs is smaller than the one of \p rhs.
+         */
+         bool operator() (const gap_t& lhs, const gap_t& rhs) const {
+             return lhs.first < rhs.first;
+         }
+     };
+
+    //!\brief The type of set to store the anchor gaps.
+    using anchor_set_type = std::set<gap_t, gap_compare<gap_t>>;
+
+    //!\brief The iterator type for an anchor set.
+    using set_iterator_type = typename anchor_set_type::iterator;
+
+    /*!\brief Helper function to compute the length of the gap indicated by the
+     * input iterator.
+     * \param[in] it    Iterator over the internal gap set.
+     * \returns The gap length corresponding to the gap pointed at by \p it.
+     *
+     * \details
+     *
+     * The length of a gap is difference of the accumulator pointed at
+     * and the one of its predecessor (if existing).
+     *
+     * \par Exceptions
+     * No-throw guarantee.
+     */
+    constexpr size_type get_gap_length(set_iterator_type it) const noexcept
+    {
+        if (it == anchors.begin())
+            return (*it).second;
+        return (*it).second - (*std::prev(it)).second;
+    }
+
+    /*!\brief Update all anchor gaps after the indicated position by adding an offset.
+     * \param[in] pos    Gap index after which to perform the update.
+     * \param[in] offset Offset to be added to the virtual gap positions and its accumulators.
+     *
+     * \details
+     *
+     * In order to avoid key conflicts when inserting into anchors, the
+     * update is done in reverse manner excluding the indicated gap.
+     *
+     * \par Complexity
+     * Linear in the number of gaps.
+     */
+    void rupdate(size_type const pos, size_type const offset)
+    {
+        for (auto it = std::prev(anchors.end(), 1); (*it).first > pos;)
+        {
+            anchors.emplace_hint(it, gap_t{(*it).first + offset, (*it).second + offset});
+            anchors.erase(*it--);
+        }
+    }
+
+    /*!\brief Update all anchor gaps after indicated position by substracting an offset.
+     * \param[in] it     Iterator pointing to the first anchor node to update.
+     * \param[in] offset Offset to be removed from the virtual gap positions and its accumulators.
+     *
+     * \details
+     *
+     * In order to avoid key conflicts when inserting into anchors, the
+     * decreasing is done in a forward manner excluding the indicated gap.
+     *
+     * \par Complexity
+     * Linear in the number of gaps.
+     *
+     * \par Exceptions
+     * Throws assert when initial update position is out of range.
+     */
+    void update(set_iterator_type it, size_type const offset)
+    {
+        while (it != anchors.end())
+        {
+            gap_t gap{(*it).first - offset, (*it).second - offset};
+            it = anchors.erase(it);
+            it = anchors.insert(it, gap);
```

Comment:
no hint here?

Reply:
There is not `insert_hint` but the `it` parameter is the hint :)